akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/BehaviorTestKitImpl.scala

```diff
@@ -132,7 +133,36 @@ private[akka] final class BehaviorTestKitImpl[T](_path: ActorPath, _initialBehav
     try {
       context.setCurrentActorThread()
       try {
-        currentUncanonical = Behavior.interpretMessage(current, context, message)
+        //we need this to handle message adapters related messages
+        val intercepted0 = Behaviors.intercept{ () =>
+          new BehaviorInterceptor[Any, T]() {
+            import BehaviorInterceptor._
+            /**
+             * Intercept a message sent to the running actor. Pass the message on to the next behavior
+             * in the stack by passing it to `target.apply`, return `Behaviors.same` without invoking `target`
+             * to filter out the message.
+             *
+             * @return The behavior for next message or signal
+             */
+            override def aroundReceive(ctx: TypedActorContext[Any], msg: Any, target: BehaviorInterceptor.ReceiveTarget[T]): Behavior[T] = {
+              msg match {
+                case AdaptWithRegisteredMessageAdapter(msgToAdapt) =>
+                  val fn = context
+                    .messageAdapters
+                    .find(_._1.isInstance(msgToAdapt))
```

Comment:
Easier to read if you avoid the indexed tuple positions, for example with `messageAdapters.collectFirst { case (clazz, adapter) if clazz.isInstance(msg) => adapter }.getOrElse(...)`